stan/math/prim/fun/inv_logit.hpp

```diff
@@ -69,28 +68,45 @@ inline double inv_logit(double a) {
  */
 struct inv_logit_fun {
   template <typename T>
-  static inline auto fun(const T& x) {
-    return inv_logit(x);
+  static inline auto fun(T&& x) {
+    return inv_logit(std::forward<T>(x));
   }
 };
 
 /**
- * Vectorized version of inv_logit().
+ * Vectorized version of inv_logit() for std::vector's containing ad types.
  *
- * @tparam T type of container
- * @param x container
+ * @tparam T type of std::vector
+ * @param x std::vector
  * @return Inverse logit applied to each value in x.
  */
-template <
-    typename T, require_not_var_matrix_t<T>* = nullptr,
-    require_all_not_nonscalar_prim_or_rev_kernel_expression_t<T>* = nullptr>
-inline auto inv_logit(const T& x) {
-  return apply_scalar_unary<inv_logit_fun, T>::apply(x);
+template <typename Container, require_ad_container_t<Container>* = nullptr,
+          require_all_not_nonscalar_prim_or_rev_kernel_expression_t<
+              Container>* = nullptr,
+          require_not_rev_matrix_t<Container>* = nullptr>
+inline auto inv_logit(Container&& x) {
+  return apply_scalar_unary<inv_logit_fun, Container>::apply(
+      std::forward<Container>(x));
 }
 
-// TODO(Tadej): Eigen is introducing their implementation logistic() of this
-// in 3.4. Use that once we switch to Eigen 3.4
-
+/**
+ * Vectorized version of inv_logit() for Eigen types.
+ *
+ * @tparam T A type of either `std::vector` whose inner type inherits from
+ * `Eigen::DenseBase` or a type that directly inherits from `Eigen::DenseBase`.
+ * The inner scalar type must not have a `var` scalar type.
+ * @param x Eigen expression
```

Comment:
This overload would also be for `std::vector` with inner types of `double` or `std::vector`